analyses/molecular-subtyping-MB/03-compare-classes.Rmd

```diff
@@ -33,15 +33,32 @@ suppressPackageStartupMessages(library(plyr))
 root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
 data_dir <- file.path(root_dir, "data")
 
+# This was created in 00-mb-select-pathology-dex.Rmd
+clinical_file <- file.path(data_dir, "pbta-histologies.tsv")
+
 # set results directory
 results_dir <- file.path(root_dir, "analyses", "molecular-subtyping-MB", "results")
+
+# read in the terms file
+terms.file <- file.path("input", "mb_subtyping_path_dx_strings.json")
+```
+
+```{r}
+# Read in the JSON file that contains the strings we'll use to include or
+# exclude samples for subtyping - see 00-HGG-select-pathology-dx
+path_dx_list <- jsonlite::fromJSON(terms.file)
 ```
 </details>
 
 <details><summary>Read input</summary>
 ```{r read_data, echo = TRUE}
 # read input
-clin <- read.delim(file.path(data_dir, "pbta-histologies.tsv"))
+subset.clin <- read.delim(clinical_file) %>% 
```

Comment:
To me, this suggests that part of the filtering step in `01` could be to output a subset version of `pbta-histologies.tsv` that can be used here so you don't have to change the filtering code in two places should you need to change the filtering code. I took some similar steps on #786 in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/786/commits/e8480cd94bda49ef6e390aea7575870ba2c4e37d and https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/786/commits/c808a512ddce6e975ff4c63e4335e78ba3faa2ba.

You'd first filter based on the criteria that make something a sample included for subtyping, output that file, and then filter to RNA-seq samples to be used for the rest of that script. Side note: I don't know that you actually need to filter to RNA-seq samples looking at the rest of the code in `analyses/molecular-subtyping-MB/01-filter-and-batch-correction.R` because there's an `intersect()` step prior to `dplyr::select()`.

Reply:
Oh yeah, I hadn't noticed that. Do we want to drop the whole thing? in `01-` and only do filter in `03-`?

Reply:
I have determined that a later step apparently does require the filtering be exact.

Reply:
I didn't follow that, but I will say that I believe you do have to filter based on the criteria we've been talking about in `01` to get the correct expression matrices committed to the repo (if you don't commit those to the repo, this will fail CI because there's not a guaranteed number of MB samples in the CI files), you just don't need `experimental_strategy == "RNA-Seq"` step there as far as I can tell and you don't want to have the filtering code in two places (`01` and `03`).

Reply:
Oh I see, the batch correction step requires the filtering to RNA-seq. Anyway, the points about not repeating oneself still stand.

Reply:
Yeah, I did some rewriting of that section see how you feel about it.

Reply:
I think it is less redundant now overall.